etcdserver/raft.go

```diff
@@ -257,7 +261,9 @@ func (r *raftNode) start(rh *raftReadyHandler) {
 						}
 					}
 					if waitApply {
-						rh.waitForApply()
+						// +1 to sync with scheduled config-change job
+						// if already applied, return immediately without blocking
+						rh.waitForApply(finished + 1)
```

Comment:
Yeah scheduler-dependent function doesn't look good in there.
We only need to block until it's trigger.

> would raftDone <- struct{}{} work here

Do you mean 
```go
// https://github.com/coreos/etcd/blob/master/etcdserver/raft.go#L243
raftDone <- struct{}{}

if waitApply {
    raftDone <- struct{}{}
}
```

Then we can just  delete the entire `waitForApply` field?